src/tactic/model_converter.cpp

```diff
@@ -76,6 +82,24 @@ class concat_star_model_converter : public concat_star_converter<model_converter
         }
         UNREACHABLE();
     }
+
+    virtual void operator()(svector<symbol> & r, unsigned goal_idx) {
+        unsigned num = this->m_c2s.size();
+        for (unsigned i = 0; i < num; i++) {
+            if (goal_idx < this->m_szs[i]) {
```

Comment:
This code is copy-pasted from `virtual void operator()(model_ref & m, unsigned goal_idx)`; to be honest, I'm not exactly sure how the interaction between the goal index and the model converters works. Two lines later, though, `i` is used as an index into `m_c2s`. Should I still change it?